pkg/clustermanager/cluster_manager.go

```diff
@@ -799,6 +802,47 @@ func (c *ClusterManager) waitForNodesReady(ctx context.Context, managementCluste
 	return nil
 }
 
+func (c *ClusterManager) waitForClustersReady(ctx context.Context, managementCluster *types.Cluster, clusterName string, labels []string, checkers ...types.ClusterReadyChecker) error {
+	readyClusters, totalClusters := 0, 0
+	policy := func(_ int, _ error) (bool, time.Duration) {
+		// TODO: configure this timeout outside the function
+		return true, 30 * time.Minute * time.Duration(totalClusters-readyClusters)
+	}
+
+	areClustersReady := func() error {
+		var err error
+		readyClusters, totalClusters, err = c.countClustersReady(ctx, managementCluster, clusterName, labels, checkers...)
+		if err != nil {
+			return err
+		}
+
+		if readyClusters != totalClusters {
+			logger.V(4).Info("Clusters are not ready yet", "total", totalClusters, "ready", readyClusters, "cluster name", clusterName)
+			return errors.New("nodes are not ready yet")
+		}
+
+		logger.V(4).Info("Clusters ready", "total", totalClusters)
+		return nil
+	}
+
+	err := areClustersReady()
+	if err == nil {
+		return nil
+	}
+
+	timeout := time.Duration(totalClusters) * c.machineMaxWait
```

Comment:
move off of using machineMaxWait here for some cluster max wait instead

Reply:
Is this a future action item? I would suggest deciding on something here rather than using machineMaxWait if possible.

Reply:
Added durations for cluster ready check